common/logger/src/struct_log.rs

```diff
@@ -176,65 +172,61 @@ impl StructuredLogEntry {
         self
     }
 
-    // TODO: Remove in favor of level (https://github.com/libra/libra/issues/5484)
-    pub fn critical(self) -> Self {
-        self.level(log::Level::Error)
-    }
-
-    pub fn json_data(mut self, key: &'static str, value: Value) -> Self {
+    pub(crate) fn json_data(mut self, key: &'static str, value: Value) -> Self {
         self.data.insert(key, value);
         self
     }
 
-    pub fn data<D: Serialize>(mut self, key: &'static str, value: D) -> Self {
-        self.data.insert(
+    /// Add a data field, with a given value, will serialize into JSON and be indexed accordingly
+    pub fn data<D: Serialize>(self, key: &'static str, value: D) -> Self {
+        self.json_data(
             key,
             serde_json::to_value(value).expect("Failed to serialize StructuredLogEntry key"),
-        );
-        self
+        )
     }
 
     /// Used for errors and other types that don't serialize well
     pub fn data_display<D: Display>(self, key: &'static str, value: D) -> Self {
         self.data(key, value.to_string())
     }
 
+    /// Add a typed data field to serialize, to ensure type and name consistency
     pub fn field<D: Serialize>(self, field: &LoggingField<D>, value: D) -> Self {
         self.data(field.0, value)
     }
 
-    #[doc(hidden)] // set from macro
-    pub fn log(&mut self, log: String) -> &mut Self {
+    /// Sets the context log line used for text logs.  This is useful for migration of text logs
+    pub fn log(mut self, log: String) -> Self {
         self.log = Some(log);
         self
     }
 
     #[doc(hidden)] // set from macro
-    pub fn pattern(&mut self, pattern: &'static str) -> &mut Self {
-        self.pattern = Some(pattern);
+    pub fn add_log(&mut self, log: String) -> &mut Self {
```

Comment:
add_* seems like an anti-pattern much like set_ or get_

Reply:
Yeah, it's an anti-pattern, but the issue was the confusion between the `&mut Self` vs `Self` returns.  Rather than rework the entire macro system, I went with this to differentiate between the two.